hazelcast/include/hazelcast/client/MixedMap.h

```diff
@@ -0,0 +1,1134 @@
+
+                std::vector<std::pair<K, V> > result;
+                for (size_t i = range.first; i < range.second; ++i) {
+                    std::pair<const K *, const V *> entry = entries[i];
+                    result.push_back(std::pair<K, V>(*entry.first, *entry.second));
+                }
+                return result;
+            }
+
+            /**
+            * Adds an index to this map for the specified entries so
+            * that queries can run faster.
+            *
+            * Let's say your map values are Employee objects.
+            *
+            *   class Employee : public serialization::Portable {
+            *       //...
+            *       private:
+            *          bool active;
+            *          int age;
+            *          std::string name;
+            *
+            *   }
+            *
+            *
+            * If you are querying your values mostly based on age and active then
+            * you should consider indexing these fields.
+            *
+            *   IMap<std::string, Employee > imap = hazelcastInstance.getMap<std::string, Employee >("employees");
+            *   imap.addIndex("age", true);        // ordered, since we have ranged queries for this field
+            *   imap.addIndex("active", false);    // not ordered, because boolean field cannot have range
+            *
+            *
+            * In the server side, Index  should either have a getter method or be public.
+            * You should also make sure to add the indexes before adding
+            * entries to this map.
+            *
+            * @param attribute attribute of value
+            * @param ordered   <tt>true</tt> if index should be ordered,
+            *                  <tt>false</tt> otherwise.
+            */
+            void addIndex(const std::string &attribute, bool ordered);
+
+            /**
+            * Applies the user defined EntryProcessor to the entry mapped by the key.
+            * Returns the the ResultType which is result of the process() method of EntryProcessor.
+            *
+            * EntryProcessor should extend either Portable or IdentifiedSerializable.
+            * Notice that map EntryProcessor runs on the nodes. Because of that, same class should be implemented in java side
+            * with same classId and factoryId.
+            *
+            * @tparam EntryProcessor type of entry processor class
+            * @tparam ResultType that entry processor will return
+            * @param entryProcessor that will be applied
+            * @param key of entry that entryProcessor will be applied on
+            * @return result of entry process.
+            */
+            template<typename K, typename ResultType, typename EntryProcessor>
+            boost::shared_ptr<ResultType> executeOnKey(const K &key, EntryProcessor &entryProcessor) {
```

Comment:
Can we return the result of EntryProcessor as TypeData ?

Reply:
Corrected.